packages/svelte-check/src/writers.ts

```diff
@@ -0,0 +1,124 @@
+import * as chalk from 'chalk';
+import { Writable } from "stream";
+import {
+    Diagnostic,
+    DiagnosticSeverity,
+} from 'vscode-languageserver-protocol';
+import { offsetAt } from 'svelte-language-server';
+
+export interface Writer {
+    start: (workspaceUri: string) => void;
+    file: (d: Diagnostic[], filename: string, text: string) => void;
+    completion: (fileCount: number, errorCount: number, warningCount: number) => void;
+    failure: (err: Error) => void;
+}
+
+export class HumanFriendlyWriter implements Writer {
+    _stream: Writable;
+    _isVerbose: boolean;
+
+    constructor(stream: Writable, isVerbose = true) {
+        this._stream = stream;
+        this._isVerbose = isVerbose;
+    }
+
+    start(workspaceUri: string) {
+        if (this._isVerbose) {
+            this._stream.write('\n');
+            this._stream.write(`Loading svelte-check in workspace: ${workspaceUri}`);
+            this._stream.write('\n');
+            this._stream.write('Getting Svelte diagnostics...\n');
+            this._stream.write('====================================\n');
+            this._stream.write('\n');
+        }
+    }
+
+    file(diagnostics: Diagnostic[], filename: string, text: string): void {
+        if (diagnostics.length > 0) {
+            this._stream.write('\n');
+            this._stream.write(`${chalk.green('File')} : ${chalk.green(filename)}\n`);
+
+            diagnostics.forEach((diagnostic) => {
+                const source = diagnostic.source ? `(${diagnostic.source})` : '';
+
+                // eslint-disable-next-line max-len
+                const position = `Line: ${diagnostic.range.start.line}, Character: ${diagnostic.range.start.character}`;
+
+                // Show some context around diagnostic range
+                const startOffset = offsetAt(diagnostic.range.start, text);
+                const endOffset = offsetAt(diagnostic.range.end, text);
+                const codePrev = chalk.cyan(text.substring(Math.max(startOffset - 10, 0), startOffset));
+                const codeHighlight = chalk.magenta(text.substring(startOffset, endOffset));
+                const codePost = chalk.cyan(text.substring(endOffset, endOffset + 10));
+                const code = codePrev + codeHighlight + codePost;
+                let msg;
+
+                if (this._isVerbose) {
+                    msg = `${diagnostic.message} ${source}\n${position}\n${chalk.cyan(code)}`;
+                }
+                else {
+                    msg = diagnostic.message;
```

Comment:
I think line and character should also be shown in the non-verbose output.

Reply:
done.